### ISSUE TITLE
Failed to connect to Reaper on Docker Desktop 4.19.0 on MacOS

### ISSUE THREAD

**Issue by rfoel** [testcontainers/testcontainers-node#563]
**Expected Behaviour**
Run tests

**Actual Behaviour**
Throws error "Failed to connect to Reaper"

**Testcontainer Logs**
````
Determining test suites to run...  testcontainers [DEBUG] Found Docker client strategy "UnixSocketStrategy" +0ms
  testcontainers [DEBUG] Testing Docker client strategy "unix:///var/run/docker.sock"... +1ms
  testcontainers [DEBUG] Fetching system info... +26ms
  testcontainers [DEBUG] Node version: v18.16.0, Platform: darwin, Arch: arm64, OS: Docker Desktop, Version: 23.0.5, Arch: aarch64, CPUs: 4, Memory: 2085826560, Compose installed: true, Compose version: 2.17.3 +395ms
  testcontainers [INFO] Using Docker client strategy "UnixSocketStrategy", Docker host "localhost" (::1, 127.0.0.1) +2ms
  testcontainers [DEBUG] Not pulling image "amazon/dynamodb-local:latest" as it already exists +9ms
  testcontainers [DEBUG] Creating new Reaper for session "6b7ebfd7b5fd"... +1ms
  testcontainers [DEBUG] Not pulling image "testcontainers/ryuk:0.4.0" as it already exists +5ms
  testcontainers [INFO] Creating container for image "testcontainers/ryuk:0.4.0"... +1ms
  testcontainers [INFO] [a88483885b6c] Created container for image "testcontainers/ryuk:0.4.0" +33ms
  testcontainers [INFO] [a88483885b6c] Starting container for image "testcontainers/ryuk:0.4.0"... +0ms
  testcontainers [INFO] [a88483885b6c] Started container for image "testcontainers/ryuk:0.4.0" +294ms
  testcontainers [DEBUG] [a88483885b6c] Waiting for container to be ready... +12ms
  testcontainers [DEBUG] [a88483885b6c] Waiting for log message "/.+ Started!/"... +1ms
  testcontainers:containers [a88483885b6c] 2023/05/09 13:50:41 Pinging Docker... +0ms
  testcontainers:containers [a88483885b6c] 2023/05/09 13:50:41 Docker daemon is available! +0ms
  testcontainers:containers [a88483885b6c] 2023/05/09 13:50:41 Starting on port 8080... +0ms
  testcontainers:containers [a88483885b6c] 2023/05/09 13:50:41 Started! +0ms
  testcontainers [INFO] [a88483885b6c] Container is ready +10ms
  testcontainers [DEBUG] [a88483885b6c] Connecting to Reaper (attempt 1) on "localhost:32770"... +1ms
  testcontainers [ERROR] Reaper a88483885b6c686828ba4dba4d1c4ad3e073a6df3548ef911fa3f5194e029337 socket error: Error: connect ECONNREFUSED ::1:32770 +3ms
  testcontainers [ERROR] [a88483885b6c] Connection to Reaper closed with error +0ms
  testcontainers [DEBUG] [a88483885b6c] Connecting to Reaper (attempt 2) on "localhost:32770"... +1s
  testcontainers [ERROR] Reaper a88483885b6c686828ba4dba4d1c4ad3e073a6df3548ef911fa3f5194e029337 socket error: Error: connect ECONNREFUSED ::1:32770 +1ms
  testcontainers [ERROR] [a88483885b6c] Connection to Reaper closed with error +0ms
  testcontainers [DEBUG] [a88483885b6c] Connecting to Reaper (attempt 3) on "localhost:32770"... +1s
  testcontainers [ERROR] Reaper a88483885b6c686828ba4dba4d1c4ad3e073a6df3548ef911fa3f5194e029337 socket error: Error: connect ECONNREFUSED ::1:32770 +7ms
  testcontainers [ERROR] [a88483885b6c] Connection to Reaper closed with error +0ms
  testcontainers [DEBUG] [a88483885b6c] Connecting to Reaper (attempt 4) on "localhost:32770"... +1s
  testcontainers [ERROR] Reaper a88483885b6c686828ba4dba4d1c4ad3e073a6df3548ef911fa3f5194e029337 socket error: Error: connect ECONNREFUSED ::1:32770 +2ms
  testcontainers [ERROR] [a88483885b6c] Connection to Reaper closed with error +1ms
  testcontainers [DEBUG] [a88483885b6c] Connecting to Reaper (attempt 5) on "localhost:32770"... +1s
  testcontainers [ERROR] Reaper a88483885b6c686828ba4dba4d1c4ad3e073a6df3548ef911fa3f5194e029337 socket error: Error: connect ECONNREFUSED ::1:32770 +2ms
  testcontainers [ERROR] [a88483885b6c] Connection to Reaper closed with error +0ms
  testcontainers [ERROR] [a88483885b6c] Failed to connect to Reaper +1ms
Error: Jest: Got error running globalSetup - /Users/rafael/Developer/finance/src/infra/test/setup/setup.js, reason: Failed to connect to Reaper
    at /Users/rafael/Developer/finance/node_modules/testcontainers/dist/src/reaper.js:134:24
    at IntervalRetryStrategy.<anonymous> (/Users/rafael/Developer/finance/node_modules/testcontainers/dist/src/retry-strategy.js:37:28)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/rafael/Developer/finance/node_modules/testcontainers/dist/src/retry-strategy.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
````

**Environment Information**
- Operating System: macOS 13.3.1
- Docker Version: 23.0.5
- Docker Desktop Version: 4.19.0
- Node version: 18.16.0
- Testcontainers version: 9.7.0

**Comment by cristianrgreco** [comment]
Thanks for raising @rfoel. There was a change in docker-desktop 4.19 which affects the socket location for MacOS. Could you please confirm that the following button is unchecked, and that checking it fixes the issue?

![image](https://github.com/testcontainers/testcontainers-node/assets/10348798/4f9256d6-327b-4da0-858c-d0caa40543c8)

**Comment by rfoel** [comment]
That worked, thanks!

**Comment by cristianrgreco** [comment]
Great! I will keep this issue open until we have a proper fix, and will pin it for others to see.

**Comment by matleh** [comment]
@cristianrgreco Thanks for your quick reply. Yes, I read the docs and did
```
export DOCKER_HOST=unix://${HOME}/.colima/default/docker.sock
export TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=/var/run/docker.sock
jest .foo.spec.js
```
which results in
```
Failed to connect to Reaper

      at node_modules/testcontainers/dist/src/reaper.js:134:24
      at IntervalRetryStrategy.<anonymous> (node_modules/testcontainers/dist/src/retry-strategy.js:37:28)
      at fulfilled (node_modules/testcontainers/dist/src/retry-strategy.js:5:58)
```

`docker ps` does work so colima is running and docker working.

**Comment by cristianrgreco** [comment]
Fixed in v9.8.0
